validator/testdata/feature_tests/forms.html

```diff
@@ -86,6 +86,14 @@
   <form method="post" action-xhr="https://cdn.ampproject.org/subscribe" target="_blank">
     <input type="submit" value="Subscribe">
   </form>
+  <!-- Invalid: form action must be a non-cdn link -->
+  <form method="post" action-xhr="https://example-com.cdn.ampproject.org/subscribe" target="_blank">
+    <input type="submit" value="Subscribe">
+  </form>
+  <!-- Valid: disallowed domain checks subdomains, not suffix -->
+  <form method="post" action-xhr="https://example-cdn.ampproject.org/subscribe" target="_blank">
+    <input type="submit" value="Subscribe">
+  </form>
```

Comment:
We tend not to enumerate over all possible test case values but instead test that the code paths are being run. However since this is important, I'll go ahead and add the second dis-allowed domain.